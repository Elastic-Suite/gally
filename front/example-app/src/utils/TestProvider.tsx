import { ReactNode } from 'react'

import { IApi, schemaContext } from 'shared'

interface IProps {
  api: IApi
  children: ReactNode
}

function TestProvider(props: IProps): JSX.Element {
  const { api, children } = props
  return <schemaContext.Provider value={api}>{children}</schemaContext.Provider>
}

export default TestProvider
