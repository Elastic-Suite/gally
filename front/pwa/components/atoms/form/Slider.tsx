import { ReactNode } from 'react'
import Box from '@mui/material/Box'
import { CustomIndicatorsNumber, Slide } from './Slider.styled'
import { FormHelperText, InputLabel, SliderProps } from '@mui/material'

import IonIcon from '~/components/atoms/IonIcon/IonIcon'

import InfoTooltip from './InfoTooltip'
import { StyledFormControl } from './InputText.styled'

export interface ISliderProps extends Omit<SliderProps, 'onChange'> {
  error?: boolean
  fullWidth?: boolean
  infoTooltip?: string
  label?: string
  margin?: 'none' | 'dense' | 'normal'
  helperText?: ReactNode
  helperIcon?: string
  onChange?: (value: number) => void
  value: number
  width?: number
  required?: boolean
}

const indicatorNumber = [-99, 0, 100]

function Slider(props: ISliderProps): JSX.Element {
  const {
    error,
    fullWidth,
    helperText,
    helperIcon,
    infoTooltip,
    label,
    margin,
    onChange,
    required,
    value,
    width,
  } = props

  function handleChange(_: Event, value: number): void {
    if (onChange) {
      onChange(value)
    }
  }
  return (
    <StyledFormControl
      error={error}
      fullWidth={fullWidth}
      margin={margin}
      variant="standard"
    >
      {Boolean(label || infoTooltip) && (
        <InputLabel shrink required={required}>
          {label}
          {infoTooltip ? <InfoTooltip title={infoTooltip} /> : null}
        </InputLabel>
      )}
      <Box
        width={!fullWidth ? width : undefined}
        sx={{ marginTop: label ? '24px' : 0 }}
      >
        <CustomIndicatorsNumber>
          {indicatorNumber.map((item) => (
            <div key={item}>{item}</div>
          ))}
        </CustomIndicatorsNumber>
        <Slide
          valueLabelDisplay="on"
          max={100}
          value={value}
          min={-99}
          onChange={handleChange}
          sx={{
            '.MuiSlider-rail': {
              '::after': {
                left: value < 0 ? `calc(50% - ${-value / 2}%)` : '50%',
                width: `${Math.abs(value) / 2}%`,
              },
            },
          }}
        />
      </Box>
      {Boolean(helperText) && (
        <FormHelperText>
          {Boolean(helperIcon) && (
            <IonIcon
              name={helperIcon}
              style={{ fontSize: 18, marginRight: 2 }}
            />
          )}
          {helperText}
        </FormHelperText>
      )}
    </StyledFormControl>
  )
}

Slider.defaultProps = {
  width: 376,
}

export default Slider
