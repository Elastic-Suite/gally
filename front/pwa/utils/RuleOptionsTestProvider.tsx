import { ReactNode } from 'react'

import ruleEngineOperator from '~/public/mocks/rule_engine_operators.json'
import sourceFields from '~/public/mocks/source_fields.json'
import sourceFieldLabels from '~/public/mocks/source_field_labels.json'
import { RuleAttributeType } from 'shared'

import RuleOptionsProvider from '~/components/stateful-providers/RuleOptionsProvider/RuleOptionsProvider'

interface IProps {
  children: ReactNode
}

const sourceFieldLabelsMap = new Map(
  sourceFieldLabels['hydra:member'].map((label) => [
    label.sourceField,
    label.label,
  ])
)

const fields = sourceFields['hydra:member'].map((field) => {
  const label = sourceFieldLabelsMap.get(`/source_fields/${field.id}`)
  return {
    id: field.id,
    code: field.code,
    label: (label || field.defaultLabel) ?? '',
    type: field.type as RuleAttributeType,
  }
})

function RuleOptionsTestProvider(props: IProps): JSX.Element {
  const { children } = props
  return (
    <RuleOptionsProvider
      catalogId={-1}
      localizedCatalogId={-1}
      fields={fields}
      ruleOperators={ruleEngineOperator}
    >
      {children}
    </RuleOptionsProvider>
  )
}

export default RuleOptionsTestProvider
