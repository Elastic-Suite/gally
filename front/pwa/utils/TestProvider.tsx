import { ReactNode } from 'react'

import { schemaContext, userContext } from '~/contexts'
import { IApi, IUser } from 'shared'

import OptionsProvider from '~/components/stateful-providers/OptionsProvider/OptionsProvider'

interface IProps {
  api: IApi
  children: ReactNode
}

const timestamp = Math.floor(Date.now() / 1000)
const expires = 8 * 60 * 60 // 8 hours

const user = {
  iat: timestamp,
  exp: timestamp + expires,
  roles: ['ROLE_ADMIN', 'ROLE_CONTRIBUTOR'],
  username: 'admin@example.com',
} as IUser

function TestProvider(props: IProps): JSX.Element {
  const { api, children } = props
  return (
    <userContext.Provider value={user}>
      <schemaContext.Provider value={api}>
        <OptionsProvider>{children}</OptionsProvider>
      </schemaContext.Provider>
    </userContext.Provider>
  )
}

export default TestProvider
