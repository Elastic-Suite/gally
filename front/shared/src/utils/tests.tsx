import { RenderOptions, render, renderHook } from '@testing-library/react'
import { ReactNode } from 'react'

import { schemaContext } from '../contexts'
import { api } from '../mocks'

export type IExtendedRenderOptions = Omit<RenderOptions, 'queries'>

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  renderOptions: IExtendedRenderOptions = {}
) {
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <schemaContext.Provider value={api}>{children}</schemaContext.Provider>
    )
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions })
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  renderOptions: IExtendedRenderOptions = {}
) {
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <schemaContext.Provider value={api}>{children}</schemaContext.Provider>
    )
  }
  return renderHook(hook, { wrapper: Wrapper, ...renderOptions })
}
