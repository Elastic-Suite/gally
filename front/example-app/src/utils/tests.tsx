import { RenderOptions, render, renderHook } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { ReactNode } from 'react'
import { api } from 'gally-admin-shared'

import TestProvider from './TestProvider'

export type IExtendedRenderOptions = Omit<RenderOptions, 'queries'>

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  renderOptions: IExtendedRenderOptions = {}
) {
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <MemoryRouter>
        <TestProvider api={api}>{children}</TestProvider>
      </MemoryRouter>
    )
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions })
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  renderOptions: IExtendedRenderOptions = {}
) {
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <MemoryRouter>
        <TestProvider api={api}>{children}</TestProvider>
      </MemoryRouter>
    )
  }
  return renderHook(hook, { wrapper: Wrapper, ...renderOptions })
}
