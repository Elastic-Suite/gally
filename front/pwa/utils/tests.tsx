import { RenderOptions, render, renderHook } from '@testing-library/react'
import { PreloadedState } from '@reduxjs/toolkit'
import { ReactNode } from 'react'
import { Bundle, IUser, api } from 'shared'

import { AppStore, RootState, setupStore } from '~/store'

import AppProvider from '~/components/stateful-providers/AppProvider/AppProvider'

import TestProvider from './TestProvider'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

const timestamp = Math.floor(Date.now() / 1000)
const expires = 8 * 60 * 60 // 8 hours

const user = {
  iat: timestamp,
  exp: timestamp + expires,
  roles: ['ROLE_ADMIN', 'ROLE_CONTRIBUTOR'],
  username: 'admin@example.com',
} as IUser

const defaultState = {
  data: {
    api,
    bundles: [Bundle.VIRTUAL_CATEGORY],
    configurations: ['https://localhost/products'],
  },
  user: {
    requestedPath: '',
    token:
      'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpYXQiOjE2NTk5NTM1NDQsImV4cCI6MTY1OTk4MjM0NCwicm9sZXMiOlsiUk9MRV9BRE1JTiIsIlJPTEVfQ09OVFJJQlVUT1IiXSwidXNlcm5hbWUiOiJhZG1pbkBleGFtcGxlLmNvbSJ9.ZNE2SHmtm9WgWEMmKGMjpcfEym2ysxC_CVcLtzlpuT3jlT0Zyz4VjoQHdTav9I7s9b2QzpeDNt3Sfrbka8IHcVb_TvZle_vII-TkQGal37RtXxKk_EMMQQlzLRO3zDWYy2I0ZPXv6iMlaSWOya4egOG2s1_tzFf4MnNNm2zEQbjzAtZ_Ij9UBMlSxK19AHvHlKU9AelA-NepjUdl6f4GxYPnyIwZw3PS294Wtffj7Kzw1ZZywDljl1xN_flg70cuO0GGkUqBNHrf7zBjPmXbAZZDAefoyQfVqB2v4GzDJfO847XymvBlHtCCjWx0ThK5afyOGL2MAAZkl-AoNjMcjA',
    user,
  },
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
