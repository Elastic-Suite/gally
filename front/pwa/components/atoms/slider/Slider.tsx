import React, { useState } from 'react'
import Box from '@mui/material/Box'
import Slider from '@mui/material/Slider'
import { styled } from '@mui/system'

const Slide = styled(Slider)(({ theme }) => ({
  '.MuiSlider-root': {
    height: '10px!important',
  },
  '.MuiSlider-rail': {
    border: '1px solid',
    borderColor: theme.palette.colors.neutral[900],
    backgroundColor: theme.palette.colors.neutral[300],
    '::after': {
      content: "''",
      position: 'absolute',
      top: 0,
      height: '100%',
      background: 'red',
      borderRadius: theme.spacing(1),
    },
  },
  '.MuiSlider-track': {
    display: 'none',
  },
  '.MuiSlider-valueLabelOpen': {
    borderRadius: theme.spacing(1),
    backgroundColor: theme.palette.colors.primary[100],
    border: '1px solid',
    borderColor: theme.palette.colors.primary[300],
    color: theme.palette.colors.neutral[900],
    paddingRight: 0,
    paddingLeft: 0,
    paddingTop: 2,
    paddingBottom: 2,
    fontSize: '14px',
    width: '34px',
    fontWeight: '600',
    top: '-5px',
    '::before': {
      display: 'none',
    },
  },
  '.MuiSlider-thumb': {
    color: theme.palette.colors.primary[400],
  },
}))

function SliderSizes(): JSX.Element {
  const [val, setVal] = useState<any>(0)

  console.log(val)
  return (
    <Box sx={{ marginTop: '80px' }} width={300}>
      <Slide
        valueLabelDisplay="on"
        defaultValue={0}
        aria-label="Default"
        max={100}
        min={-99}
        onChange={(event: any) => setVal(event.target.value)}
        sx={{
          '.MuiSlider-rail': {
            '::after': {
              left:
                val <= 0
                  ? `calc(50% - ${val.toString().split('-')[1] / 2}%)`
                  : '50%',
              width:
                val <= 0
                  ? `${val.toString().split('-')[1] / 2}%`
                  : `${val / 2}%`,
            },
          },
        }}
      />
    </Box>
  )
}

export default SliderSizes
