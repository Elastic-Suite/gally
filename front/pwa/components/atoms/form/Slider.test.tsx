import { renderWithProviders } from '~/utils/tests'

import Slider from './Slider'

describe('Slider match snapshot', () => {
  it('Slider Full', () => {
    const { container } = renderWithProviders(
      <Slider
        value={9}
        label="Boost value (%)"
        infoTooltip="boost value ..."
        fullWidth={false}
        width={376}
        helperText="helperText"
        helperIcon="close"
        margin="dense"
        required
      />
    )
    expect(container).toMatchSnapshot()
  })
  it('Slider Min', () => {
    const { container } = renderWithProviders(<Slider value={9} />)
    expect(container).toMatchSnapshot()
  })
})
