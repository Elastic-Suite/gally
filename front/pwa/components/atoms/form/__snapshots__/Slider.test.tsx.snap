// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider match snapshot Slider Full 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-marginDense css-1m41rkg-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required css-ssr94h-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      Boost value (%)
      <span
        aria-label="boost value ..."
        class=""
        data-mui-internal-clone-element="true"
        style="display: inline-block; padding-left: 10px; margin-bottom: -5px;"
      >
        <ion-icon
          name="information-circle-outline"
          style="color: rgb(129, 135, 185); width: 20px; height: 20px; display: flex;"
        />
      </span>
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ufxisk-MuiFormLabel-asterisk"
      >
        â€‰
        *
      </span>
    </label>
    <div
      class="MuiBox-root css-15j7md6"
    >
      <div
        class="css-q0vrqn"
      >
        <div>
          -99
        </div>
        <div>
          0
        </div>
        <div>
          100
        </div>
      </div>
      <span
        class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-vts272-MuiSlider-root"
      >
        <span
          class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
        />
        <span
          class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
          style="left: 0%; width: 54.2713567839196%;"
        />
        <span
          class="MuiSlider-thumb MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium css-1ugfhow-MuiSlider-thumb"
          data-focusvisible="false"
          data-index="0"
          style="left: 54.2713567839196%;"
        >
          <input
            aria-orientation="horizontal"
            aria-valuemax="100"
            aria-valuemin="-99"
            aria-valuenow="9"
            data-index="0"
            max="100"
            min="-99"
            step="1"
            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
            type="range"
            value="9"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-valueLabelOpen MuiSlider-valueLabel css-thdgn5-MuiSlider-valueLabel"
          >
            <span
              class="MuiSlider-valueLabelCircle"
            >
              <span
                class="MuiSlider-valueLabelLabel"
              >
                9
              </span>
            </span>
          </span>
        </span>
      </span>
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-sizeMedium css-1riwtgr-MuiFormHelperText-root"
    >
      <ion-icon
        name="close"
        style="font-size: 18px; margin-right: 2px;"
      />
      helperText
    </p>
  </div>
</div>
`;

exports[`Slider match snapshot Slider Min 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1pl6002-MuiFormControl-root"
  >
    <div
      class="MuiBox-root css-plopzj"
    >
      <div
        class="css-q0vrqn"
      >
        <div>
          -99
        </div>
        <div>
          0
        </div>
        <div>
          100
        </div>
      </div>
      <span
        class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-vts272-MuiSlider-root"
      >
        <span
          class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
        />
        <span
          class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
          style="left: 0%; width: 54.2713567839196%;"
        />
        <span
          class="MuiSlider-thumb MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium css-1ugfhow-MuiSlider-thumb"
          data-focusvisible="false"
          data-index="0"
          style="left: 54.2713567839196%;"
        >
          <input
            aria-orientation="horizontal"
            aria-valuemax="100"
            aria-valuemin="-99"
            aria-valuenow="9"
            data-index="0"
            max="100"
            min="-99"
            step="1"
            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
            type="range"
            value="9"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-valueLabelOpen MuiSlider-valueLabel css-thdgn5-MuiSlider-valueLabel"
          >
            <span
              class="MuiSlider-valueLabelCircle"
            >
              <span
                class="MuiSlider-valueLabelLabel"
              >
                9
              </span>
            </span>
          </span>
        </span>
      </span>
    </div>
  </div>
</div>
`;
